### PR TITLE
integrating Plutus in Ledger code

### PR DESCRIPTION
**Pull request by TimSheard**
Jared, Alex and I, have gotten the Plutus package to compile with the Ledger code.
This required adjusting some nix-files, some cabal files, and altering the version requirements
of some of the dependencies. 

To demonstrate this capability, I have integrated Language.PlutusTx.Data into Cardano.Alonzo.Data.
Wrote some instances, a pretty printer, and an Arbitrary instance. This might be a good place
to push back to master, as the changes touch very few files.

All seems to be working, except when I try to run the ormolise script, where I get this error message. Probably we did something bad to the nix files.

### CODE REVIEW

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs [hunk 1]
```diff
@@ -56,42 +59,51 @@ import Data.MemoBytes (Mem, MemoBytes (..), memoBytes)
 import Data.Set (Set)
 import Data.Typeable (Typeable)
 import GHC.Generics (Generic)
+-- import Plutus.V1.Ledger.Scripts
```
Comment (JaredCorduan) [inline alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs:62]:
```suggestion
import Plutus.V1.Ledger.Scripts ()
```

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs [hunk 2]
```diff
@@ -56,42 +59,51 @@ import Data.MemoBytes (Mem, MemoBytes (..), memoBytes)
...
 import Shelley.Spec.Ledger.Metadata (Metadata)
 
--- =====================================================================
--- PlutusData is a placeholder for the type that Plutus expects as data.
-
-data PlutusData = NotReallyData
-  deriving (Eq, Show, Ord, Generic)
-
-instance NoThunks PlutusData
+-- import qualified Shelley.Spec.Ledger.Metadata as Ledger
```
Comment (JaredCorduan) [inline alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs:67]:
```suggestion
```

File: cabal.project [hunk 3]
```diff
@@ -1,4 +1,4 @@
-index-state: 2020-11-15T00:00:00Z
+index-state: 2021-02-01T00:00:00Z
```
Comment (JaredCorduan) [inline cabal.project]: @nc6 is this going to cause any problems?

**Review by JaredCorduan — APPROVED** [review]
looks good to me!

File: cabal.project (hunk 3 above)
Comment (nc6) [reply cabal.project]: Not now we've bumped haskell.nix

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs [hunk 4]
```diff
@@ -56,42 +59,51 @@ import Data.MemoBytes (Mem, MemoBytes (..), memoBytes)
...
+      encPlutus (Plutus.List xs) = Sum Plutus.List 2 !> toList xs
+      encPlutus (Plutus.I i) = Sum Plutus.I 3 !> To i
+      encPlutus (Plutus.B bytes) = Sum Plutus.B 4 !> To bytes
+
+deriving instance NoThunks Plutus.Data
 
 -- ============================================================================
 -- the newtype Data is a wrapper around the type that Plutus expects as data.
 -- The newtype will memoize the serialized bytes. The strategy is to replace
 -- PlutusData  with the correct type
```
Comment (nc6) [inline alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs]:
```suggestion
-- The newtype will memoize the serialized bytes.
```

**Review by nc6 — APPROVED** [review]
LGTM

**Pull request merged**